modules/core/src/v2/coins/xlm.ts

```diff
@@ -459,52 +460,50 @@ export class Xlm extends BaseCoin {
   }
 
   /**
-   * Attempt to resolve a stellar address into a stellar account
-   * If address domain matches bitgo's then resolve on our federation server
-   * Else, make the request to the federation server hosting the address
+   * Perform federation lookups
+   * Our federation server handles lookups for bitgo as well as for other federation domains
    *
-   * @param address - stellar address to look for
+   * @param {String} [address] - address to look up
+   * @param {String} [accountId] - account id to look up
    */
-  federationLookupByName(address: string): Bluebird<stellar.FederationServer.Record> {
+  federationLookup({ address, accountId }: { address?: string, accountId?: string }): Bluebird<stellar.FederationServer.Record> {
     const self = this;
     return co<stellar.FederationServer.Record>(function *() {
-      const addressParts = _.split(address, '*');
-      if (addressParts.length !== 2) {
-        throw new InvalidAddressError(`invalid stellar address: ${address}`);
-      }
-      const [, homeDomain] = addressParts;
       try {
-        if (homeDomain === self.homeDomain) {
-          const federationServer = self.getBitGoFederationServer();
+        const federationServer = self.getBitGoFederationServer() as stellar.FederationServer;
```

Comment:
Why do we need the `as stellar.FederationServer` cast? Doesn't `getBitGoFederationServer` return this type?